nomenclature/config.py

```diff
@@ -41,6 +41,19 @@ def repository_dimension_path(self) -> str:
 
 class RegionCodeListConfig(CodeListConfig):
     country: bool = False
+    nuts: dict[str, str | list[str]] = None
+
+    @field_validator("nuts")
+    @classmethod
+    def check_nuts(
+        cls, v: dict[str, str | list[str]] | None
+    ) -> dict[str, str | list[str]] | None:
+        if v and not all(k in ["nuts-1", "nuts-2", "nuts-3"] for k in v.keys()):
+            raise ValueError(
+                "Invalid fields for `nuts` in configuration. "
+                "NUTS level must be 1, 2 and/or 3."
```

Comment:
That might be somewhat ambiguous, I'd make it explicit:
```suggestion
                "Allowed values are: 'nuts-1', 'nuts-2' and 'nuts-3'."
```